File: common/src/main/java/me/kvalbrus/multibans/common/punishment/punishments/MultiTemporaryChatMute.java
package me.kvalbrus.multibans.common.punishment.punishments;

import java.util.List;
import me.kvalbrus.multibans.api.punishment.creator.PunishmentCreator;
import me.kvalbrus.multibans.api.punishment.target.PunishmentTarget;
import me.kvalbrus.multibans.api.punishment.PunishmentType;
import me.kvalbrus.multibans.api.punishment.punishments.TemporaryChatMute;
import me.kvalbrus.multibans.common.managers.PluginManager;
import me.kvalbrus.multibans.common.permissions.Permission;
import me.kvalbrus.multibans.common.punishment.MultiTemporaryPunishment;
import me.kvalbrus.multibans.common.utils.Message;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class MultiTemporaryChatMute extends MultiTemporaryPunishment implements TemporaryChatMute {

    public MultiTemporaryChatMute(@NotNull PluginManager pluginManager,
                                  @NotNull String id,
                                  @NotNull PunishmentTarget target,
                                  @NotNull PunishmentCreator creator,
                                  long createdDate,
                                  long startedDate,
                                  long duration,
                                  @Nullable String reason,
                                  @Nullable String comment,
                                  @Nullable PunishmentCreator cancellationCreator,
                                  long cancellationDate,
                                  @Nullable String cancellationReason,
                                  @NotNull List<String> servers, boolean cancelled) {
        super(pluginManager, PunishmentType.TEMP_MUTE, id, target, creator, createdDate, startedDate,
            duration, reason, comment, cancellationCreator, cancellationDate, cancellationReason,
            servers, cancelled);
    }

    @NotNull
    @Override
    public String getActivateMessageForListener() {
        return Message.TEMPMUTECHAT_ACTIVATE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public String getActivateMessageForExecutor() {
        return Message.TEMPMUTECHAT_ACTIVATE_EXECUTOR.getMessage();
    }

    @Nullable
    @Override
    public String getActivateMessageForTarget() {
        return Message.TEMPMUTECHAT_ACTIVATE_TARGET.getMessage();
    }

    @NotNull
    @Override
    public String getDeactivateMessageForListener() {
        return Message.TEMPMUTECHAT_DEACTIVATE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public String getDeactivateMessageForExecutor() {
        return Message.TEMPMUTECHAT_DEACTIVATE_EXECUTOR.getMessage();
    }

    @Nullable
    @Override
    public String getDeactivateMessageForTarget() {
        return Message.TEMPMUTECHAT_DEACTIVATE_TARGET.getMessage();
    }

    @NotNull
    @Override
    public String getDeleteMessageForListener() {
        return Message.TEMPMUTECHAT_DELETE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public String getDeleteMessageForExecutor() {
        return Message.TEMPMUTECHAT_DELETE_EXECUTOR.getMessage();
    }

    @Nullable
    @Override
    public String getDeleteMessageForTarget() {
        return Message.TEMPMUTECHAT_DELETE_TARGET.getMessage();
    }

    @NotNull
    @Override
    public String getReasonChangeMessageForExecutor() {
        return Message.TEMPMUTECHAT_REASON_CREATE_CHANGE_EXECUTOR.getMessage();
    }

    @NotNull
    @Override
    public String getReasonChangeMessageForListener() {
        return Message.TEMPMUTECHAT_REASON_CREATE_CHANGE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public String getCommentChangeMessageForExecutor() {
        return Message.TEMPMUTECHAT_COMMENT_CHANGE_EXECUTOR.getMessage();
    }

    @NotNull
    @Override
    public String getCommentChangeMessageForListener() {
        return Message.TEMPMUTECHAT_COMMENT_CHANGE_LISTEN.getMessage();
    }

    @NotNull
    @Override
    public Permission getPermissionForListener() {
        return Permission.PUNISHMENT_TEMPMUTECHAT_LISTEN;
    }
}

File: common/src/main/kotlin/me/kvalbrus/multibans/common/punishment/punishments/MultiTemporaryChatMute.kt
package me.kvalbrus.multibans.common.punishment.punishments

import me.kvalbrus.multibans.api.punishment.action.Action
import me.kvalbrus.multibans.api.punishment.action.CreationAction
import me.kvalbrus.multibans.api.punishment.punishments.PunishmentType
import me.kvalbrus.multibans.api.punishment.punishments.TemporaryChatMute
import me.kvalbrus.multibans.common.managers.PluginManager
import me.kvalbrus.multibans.common.permissions.Permission
import me.kvalbrus.multibans.common.utils.Message.*
import me.kvalbrus.multibans.common.punishment.MultiTemporaryPunishment
import me.kvalbrus.multibans.common.utils.Message

final class MultiTemporaryChatMute : MultiTemporaryPunishment, TemporaryChatMute {

    constructor(
        pluginManager: PluginManager,
        id: String,
        creationAction: CreationAction,
        activations: MutableList<Action> = mutableListOf(),
        deactivations: MutableList<Action> = mutableListOf(),
        startedDate: Long,
        duration: Long,
        comment: String,
        servers: List<String>,
        cancelled: Boolean) : super(pluginManager, PunishmentType.TEMP_MUTE, id, creationAction,
        activations, deactivations, startedDate, duration, comment, servers, cancelled)

    override val activateMessageForListener: Message = TEMPMUTECHAT_ACTIVATE_LISTEN
    override val activateMessageForExecutor: Message = TEMPMUTECHAT_ACTIVATE_EXECUTOR
    override val activateMessageForTarget: Message = TEMPMUTECHAT_ACTIVATE_TARGET
    override val deactivateMessageForListener: Message = TEMPMUTECHAT_DEACTIVATE_LISTEN
    override val deactivateMessageForExecutor: Message = TEMPMUTECHAT_DEACTIVATE_EXECUTOR
    override val deactivateMessageForTarget: Message = TEMPMUTECHAT_DEACTIVATE_TARGET
    override val createMessageForListener: Message = TEMPMUTECHAT_ACTIVATE_LISTEN
    override val createMessageForExecutor: Message = TEMPMUTECHAT_ACTIVATE_EXECUTOR
    override val createMessageForTarget: Message = TEMPMUTECHAT_ACTIVATE_TARGET
    override val deleteMessageForListener: Message = TEMPMUTECHAT_DELETE_LISTEN
    override val deleteMessageForExecutor: Message = TEMPMUTECHAT_DELETE_EXECUTOR
    override val deleteMessageForTarget: Message = TEMPMUTECHAT_DELETE_TARGET
    override val reasonChangeMessageForExecutor: Message = TEMPMUTECHAT_REASON_CREATE_CHANGE_EXECUTOR
    override val reasonChangeMessageForListener: Message = TEMPMUTECHAT_REASON_CREATE_CHANGE_LISTEN
    override val commentChangeMessageForExecutor: Message = TEMPMUTECHAT_COMMENT_CHANGE_EXECUTOR
    override val commentChangeMessageForListener: Message = TEMPMUTECHAT_COMMENT_CHANGE_LISTEN
    override val permissionForListener: Permission = Permission.PUNISHMENT_TEMPMUTECHAT_LISTEN
}